### PR TITLE
Add additional metrics

### PR DESCRIPTION
**Pull request by penalosa**
This expands metrics collection to:
* Detect Pages & Workers CI
* Filter out default args (e.g. --x-versions, --x-dev-env, and --latest) by only including args that were in the user's `argv`

---

- Tests
  - [ ] TODO (before merge)
  - [x] Tests included
  - [ ] Tests not necessary because:
- E2E Tests CI Job required? (Use "e2e" label or ask maintainer to run separately)
  - [ ] I don't know
  - [ ] Required
  - [x] Not required because: covered by unit tests
- Public documentation
  - [ ] TODO (before merge)
  - [ ] Cloudflare docs PR(s): 
  - [x] Documentation not necessary because: covered by existing telemetry docs

### CODE REVIEW

File: packages/wrangler/turbo.json [hunk 1]
```diff
@@ -36,6 +36,7 @@
 				"EXPERIMENTAL_MIDDLEWARE",
 				"FORMAT_WRANGLER_ERRORS",
 				"CF_PAGES",
```
Comment (CarmenPopoviciu) [inline packages/wrangler/turbo.json:38]: I wish we could rename this to `PAGES_CI`

File: packages/wrangler/src/is-ci.ts [hunk 2]
```diff
@@ -9,6 +9,8 @@ import isCI from "is-ci";
 export const CI = {
 	/** Is Wrangler currently running in a CI? */
 	isCI() {
-		return isCI;
+		return (
+			isCI || process.env.CF_PAGES === "1" || process.env.WORKERS_CI === "1"
```
Comment (CarmenPopoviciu) [inline packages/wrangler/src/is-ci.ts:13]: https://github.com/cloudflare/workers-sdk/pull/7549/files#r1886684817

like here for example

**Review by CarmenPopoviciu — APPROVED** [review]

File: packages/wrangler/src/__tests__/metrics.test.ts [hunk 3]
```diff
@@ -343,6 +339,50 @@ describe("metrics", () => {
...
+				expect(std.debug).toContain('"command":"wrangler login"');
+			});
+
+			it("should include args provided by the user", async () => {
+				const requests = mockMetricRequest();
+
+				await runWrangler("docs arg --search 'some search term'");
+
+				expect(requests.count).toBe(2);
+				// Notably, this _doesn't_ include default args (e.g. --x-versions)
```
Comment (vicb) [inline packages/wrangler/src/__tests__/metrics.test.ts:382]: nit: `expect(std.debug).not.toContain(...)` ?

File: packages/wrangler/src/is-interactive.ts [hunk 4]
```diff
@@ -1,13 +1,13 @@
...
+import { CI, isPagesCI, isWorkersCI } from "./is-ci";
 
 /**
  * Test whether the process is "interactive".
  * Reasons it may not be interactive: it could be running in CI,
  * or you're piping values from / to another process, etc
  */
 export default function isInteractive(): boolean {
-	if (process.env.CF_PAGES === "1") {
+	if (isPagesCI() || isWorkersCI()) {
```
Comment (vicb) [inline packages/wrangler/src/is-interactive.ts:10]: q: without knowing the context I would use `CI.isCI()`. Maybe add a comment on why it is not used.

**Pull request merged**